hivemind/utils/tensor_descr.py

```diff
@@ -70,13 +83,55 @@ def from_tensor(cls, tensor: torch.Tensor, compression=CompressionType.NONE) ->
             device=tensor.device,
             requires_grad=tensor.requires_grad,
             pin_memory=_safe_check_pinned(tensor),
-            compression=compression if tensor.is_floating_point() else CompressionType.NONE
+            compression=compression if tensor.is_floating_point() else CompressionType.NONE,
         )
 
     def make_empty(self, *batch_size: int, **kwargs) -> torch.Tensor:
         assert self.shape[0] is None, "Make sure 0-th dimension is not specified (set to None)"
         return super().make_empty(size=(*batch_size, *self.shape[1:]), **kwargs)
 
+    def packb(self) -> bytes:
+        obj_dict = asdict(self)
+        obj_dict.pop("device")
+
+        if self.dtype is not None:
+            obj_dict["dtype"] = str(self.dtype)
+        else:
+            obj_dict.pop("dtype")
```

Comment:
```suggestion
```

I suggest to remove this (= keep `obj_dict["dtype"] = None` in this case) and replace `if "dtype" in obj_dict:` with the `if obj_dict["dtype"] is not None` line in `unpack()`.

If we do that for dtype, layout, etc., the code will become shorter and we will have a way to distinguish the None value from the absence of the field (may be useful if this class is extended in future).